src/main/java/fr/cnes/sonarqube/plugins/framac/sensor/FramaCSensor.java

```diff
@@ -0,0 +1,283 @@
+            InputFile inputFile = fileSystem.inputFile(predicate);
+            if(inputFile!=null) {
+                result.put(file.getFilePath(), inputFile);
+            } else {
+                LOGGER.warn(String.format(
+                        "The source file '%s' was not found.",
+                        file.getFilePath()
+                ));
+            }
+        }
+
+        return result;
+    }
+
+    /**
+     * Returns a list of processable result file's path.
+     *
+     * @param config Configuration of the analysis where properties are put.
+     * @param fileSystem The current file system.
+     * @return Return a list of path 'findable' in the file system.
+     */
+    private List<String> getReportFiles(final Configuration config, final FileSystem fileSystem) {
+        // Contains the result to be returned.
+        final List<String> result = new ArrayList<>();
+
+        // Retrieves the non-verified path list from the SonarQube property.
+        final String[] pathArray = config.getStringArray(FramaCPluginProperties.REPORT_PATH_KEY);
+
+        // Check if each path is known by the file system and add it to the processable path list,
+        // otherwise print a warning and ignore this result file.
+        for(String path : pathArray) {
+            final File file = new File(fileSystem.baseDir(), path);
+            if(file.exists() && file.isFile()) {
+                result.add(path);
+                LOGGER.info(String.format("Results file %s has been found and will be processed.", path));
+            } else {
+                LOGGER.warn(String.format("Results file %s has not been found and wont be processed.", path));
+            }
+        }
+
+        return result;
+    }
+
+    /**
+     * Check if a rule is activated in current analysis.
+     *
+     * @param activeRules Set of active rules during an analysis.
+     * @param rule Key (Frama-C) of the rule to check.
+     * @return True if the rule is active and false if not or not exists.
+     */
+    private boolean isRuleActive(final ActiveRules activeRules, final String rule) {
+        final RuleKey ruleKey = RuleKey.of(FramaCRulesDefinition.getRepositoryKeyForLanguage(), rule);
+        return activeRules.find(ruleKey)!=null;
+    }
+
+    /**
+     *  Analyze a report file provided by the external tool Frama-C.
+     *  Check the report file integrity (exist, not empty and readable).
+     *
+     * @param context SonarQube sensor context.
+     * @param resultFile File containing issues in csv format.
+     */
+    private List<FramaCError> analyseReportCsv(
+            final SensorContext context,
+            final String resultFile) {
+
+        List<FramaCError> errors = null;
+
+        try {
+            if (null != resultFile) {
+                final Path path = Paths.get(resultFile);
+                try (FileChannel reportFile = FileChannel.open(path)) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Extract this nested try block into a separate method. ([squid:S1141](https://sonarcloud.io/organizations/lequal/rules#rule_key=squid%3AS1141))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=fr.cnes.sonarqube.plugins%3Asonar-frama-c-plugin&pullRequest=20&issues=AWXs1bJoxJR_uy9B9CnS&open=AWXs1bJoxJR_uy9B9CnS)

[ISSUE_KEY]: AWXs1bJoxJR_uy9B9CnS